File: Next Greater Node In Linked List.cpp
class Solution {
public:
    vector<int> nextLargerNodes(ListNode* head) {
        vector<int> answer;
        ListNode *temp1 = head;
        ListNode *temp2 = head->next;
        while(temp1 != NULL){
            temp2 = temp1->next;
            while(temp2 != NULL){
                if(temp2->val > temp1->val){
                    answer.push_back(temp2->val);
                    break;
                }
                if(temp2->next == NULL)
                    answer.push_back(0);
                temp2 = temp2->next;
            }
            if(temp1->next == NULL)
                    answer.push_back(0);
            temp1 = temp1->next;
        }
        return answer;
    }
};

File: Next Greater Node In Linked List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def nextLargerNodes(self, head: Optional[ListNode]) -> List[int]:
        nums = []
        stack = []

        while head:
            nums.append(head.val)
            head = head.next
        answer = [0] * len(nums)

        for i in range(len(nums)):
            while stack and nums[stack[-1]] < nums[i]:
                answer[stack.pop()] = nums[i]
            stack.append(i)
        
        return answer
